calcom/cal.com
# chore: Refactor `getLocationGroupedOptions`

## What does this PR do?



This is a stacked PR on top of enabling conferencing apps for teams #10383

This PR refactors where `getLocationGroupedOptions` is used 

Fixes # (issue)



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



- Install a video app
- Go to the event type setup page and open the edit location dialog
    - All locations should be loaded
- On that page remove the location so it's blank
    - When adding a new location, all locations should be loaded
- Create a booking
    - Change the location of the booking to open the edit location dialog
    - All locations should be loaded

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
